lib/vuln/subdomain_takeover/engine.py

```diff
@@ -0,0 +1,399 @@
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sys.stdout.flush()
+        s.settimeout(timeout_sec)
+        s.connect((targ, port))
+        return s
+    except Exception as e:
+        return None
+
+
+def sub_takeover(
+    target,
+    port,
+    timeout_sec,
+    log_in_file,
+    language,
+    time_sleep,
+    thread_tmp_filename,
+    socks_proxy,
+    scan_id,
+    scan_cmd,
+    extra_requirement,
+):
+    try:
+        s = conn(target, port, timeout_sec, socks_proxy)
+        if not s:
+            return False
+        else:
+            if target_type(target) != "HTTP" and port == 443:
+                target = "https://" + target
+            if target_type(target) != "HTTP" and port == 80:
+                target = "http://" + target
+            headers = {
+                "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome 63.0.3239.132 Safari/537.36",
+                "Accept": "text/html,application/xhtml+xml,application/xml; q=0.9,image/webp,image/apng,*/*;q=0.8",
+                "Accept-Language": "en-US,en;q=0.9",
+                "Accept-Encoding": "gzip, deflate, br",
+            }
+            req = requests.get(
+                target, headers=headers, verify=False, timeout=timeout_sec
+            )
+            for key, value in extra_requirement.items():
+                if (
+                    key != "sub_takeover_vuln_ports"
+                    and value[0].lower() in req.text.lower()
+                ):
+                    return True
+            return False
+    except Exception as e:
+        # some error warning
+        return False
+
+
+def __sub_takeover(
+    target,
+    port,
+    timeout_sec,
+    log_in_file,
+    language,
+    time_sleep,
+    thread_tmp_filename,
+    socks_proxy,
+    scan_id,
+    scan_cmd,
+    extra_requirement,
+):
+    if sub_takeover(
+        target,
+        port,
+        timeout_sec,
+        log_in_file,
+        language,
+        time_sleep,
+        thread_tmp_filename,
+        socks_proxy,
+        scan_id,
+        scan_cmd,
+        extra_requirement,
+    ):
+        info(
+            messages(language, "target_vulnerable").format(
+                target,
+                port,
+                "Subdomain Takeover Vulnerability found which will allow an adversary to claim and take control of the victim's subdomain.",
+            )
+        )
+        __log_into_file(thread_tmp_filename, "w", "0", language)
+        data = json.dumps(
+            {
+                "HOST": target,
+                "USERNAME": "",
+                "PASSWORD": "",
+                "PORT": port,
+                "TYPE": "subomain_takeover_vuln",
+                "DESCRIPTION": messages(language, "vulnerable").format(
+                    "Subdomain Takeover Vulnerability found which will allow an adversary to claim and take control of the victim's subdomain."
```

Comment:
It is not 100% certain that takeover vulnerability  so please change the text to "Potential Subdomain Takeover Vulnerability found" and list the pattern name found (e.g. heroku /uservoice  etc...)

Reply:
Changed as suggested